groupease-server/src/main/java/io/github/groupease/group/GroupWebService.java

```diff
@@ -0,0 +1,93 @@
+package io.github.groupease.group;
+
+import javax.annotation.Nonnull;
+import javax.ws.rs.*;
+import javax.ws.rs.core.MediaType;
+
+import com.codahale.metrics.annotation.Timed;
+import javax.inject.Inject;
+
+import io.github.groupease.model.Group;
+import io.github.groupease.model.Member;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.invoke.MethodHandles;
+import java.util.List;
+
+/**
+ * REST-ful web service for {@link Group} instances.
+ */
+@Path("groups")
+@Produces(MediaType.APPLICATION_JSON)
+public class GroupWebService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    private final GroupDao groupDao;
+
+    @Inject
+    public GroupWebService(@Nonnull GroupDao groupDao)
+    {
+        this.groupDao = groupDao;
+    }
+
+    public List<Group> listInChannel(long channelId)
+    {
+        LOGGER.debug("GroupWebService.listInChannel({})", channelId);
+        //Todo: auth check that caller is member of channel
```

Comment:
These sorts of checks should go in the missing service layer for separation of concerns and easy unit testing.